third-party/realdds/src/dds-device-impl.cpp

```diff
@@ -94,12 +94,15 @@ void dds_device::impl::run( size_t message_timeout_ms )
     _running = true;
 
     // Start handling options only after init() is done
+    if( !_notifications_reader )
```

Comment:
Not sure what this is doing here... belongs inside `create_notifications_reader()`?

Reply:
It is a check before dereferencing `_notification_reader` in the next line.